src/daemon/bitcoind/actions.rs

```diff
@@ -601,16 +546,98 @@ pub fn bitcoind_main_loop(
             continue;
         }
 
-        // When bitcoind isn't synced, poll each 30s
         if let Some(last_poll) = last_poll {
             if now.duration_since(last_poll) < poll_interval {
-                thread::sleep(poll_interval - now.duration_since(last_poll));
+                thread::sleep(Duration::from_millis(500));
                 continue;
             }
         }
 
         last_poll = Some(now);
-        update_deposits(&mut revaultd, &bitcoind, &mut deposits_cache)?;
-        update_tip(&mut revaultd, &bitcoind)?;
+        update_deposits(
+            &mut revaultd,
+            &bitcoind.read().unwrap(),
+            &mut deposits_cache,
+        )?;
+        update_tip(&mut revaultd, &bitcoind.read().unwrap())?;
     }
+
+    Ok(())
+}
+
+fn wallet_transaction(bitcoind: &BitcoinD, txid: Txid) -> Option<WalletTransaction> {
+    let res = bitcoind.get_wallet_transaction(txid);
+    if let Ok((hex, blockheight, received_time)) = res {
+        Some(WalletTransaction {
+            hex,
+            blockheight,
+            received_time,
+        })
+    } else {
+        log::trace!(
+            "Got '{:?}' from bitcoind when requesting wallet transaction '{}'",
+            res,
+            txid
+        );
+        None
+    }
+}
```

Comment:
```rust
if res.is_err() {
    log::trace!(
            "Got '{:?}' from bitcoind when requesting wallet transaction '{}'",
            res,
            txid
        );
}

res.map_or_else(|(hex, blockheight, received_time)| Some(WalletTransaction{hex, blockheight, received_time}), None)
```

Reply:
Meh.. This was not changed and i'm not sure it makes it clearer (maybe more clever, but not sure about clearer :p)

Reply:
ok